python/ray/dag/class_node.py

```diff
@@ -230,3 +230,99 @@ def num_returns(self) -> int:
             method = self._get_remote_method(self._method_name)
             num_returns = method.__getstate__()["num_returns"]
         return num_returns
+
+
+@DeveloperAPI
```

Comment:
should we specify `ClassMethodOutputNode` as a DeveloperAPI? The definition of DeveloperAPI is:

```
 Developer APIs are lower-level methods explicitly exposed to advanced Ray
 users and library developers. Their interfaces may change across minor
 Ray releases.
```

I can't think of a situation where Ray users or library developers would need to call the method directly. If you decide to remove the `DeveloperAPI` annotation, you can change the class name from `ClassMethodOutputName` to `_ClassMethodOutputNode` to avoid failing the linter.

Reply:
Ack on not calling the method directly. I'm plannning to merge `ClassMethodOutputNode` into `ClassMethodNode`.

Reply:
Done.